api/data_refinery_api/views.py

```diff
@@ -606,34 +609,99 @@ def get(self, request, format=None):
-
-        data['processor_jobs'] = {}
-        data['processor_jobs']['total'] = ProcessorJob.objects.count()
-        data['processor_jobs']['pending'] = ProcessorJob.objects.filter(start_time__isnull=True).count()
-        data['processor_jobs']['completed'] = ProcessorJob.objects.filter(end_time__isnull=False).count()
-        data['processor_jobs']['open'] = ProcessorJob.objects.filter(start_time__isnull=False, end_time__isnull=True).count()
-        data['processor_jobs']['average_time'] = ProcessorJob.objects.filter(start_time__isnull=False, end_time__isnull=False).aggregate(average_time=Avg(F('end_time') - F('start_time')))['average_time']
+        data['survey_jobs'] = self._get_job_stats(SurveyJob.objects, range_param)
+        data['downloader_jobs'] = self._get_job_stats(DownloaderJob.objects, range_param)
+        data['processor_jobs'] = self._get_job_stats(ProcessorJob.objects, range_param)
+        data['samples'] = self._get_object_stats(Sample.objects, range_param)
+        data['experiments'] = self._get_object_stats(Experiment.objects, range_param)
 
         return Response(data)
 
+    def _get_job_stats(self, jobs, range_param):
+        result = {
+            'total': jobs.count(),
+            'pending': jobs.filter(start_time__isnull=True).count(),
+            'completed': jobs.filter(end_time__isnull=False).count(),
+            'open': jobs.filter(start_time__isnull=False, end_time__isnull=True).count(),
+            # via https://stackoverflow.com/questions/32520655/get-average-of-difference-of-datetime-fields-in-django
+            'average_time': jobs.filter(start_time__isnull=False, end_time__isnull=False).aggregate(
+                average_time=Avg(F('end_time') - F('start_time')))['average_time']
+        }
+
+        if result['average_time'] is None:
+            result['average_time'] = 0
+        else:
+            result['average_time'] = result['average_time'].total_seconds()
+
+        if range_param is not None:
+            result['timeline'] = self._jobs_timeline(jobs, range_param)
+
+        return result
+
+    def _get_object_stats(self, objects, range_param):
+        result = {
+            'total': objects.count()
+        }
+
+        if range_param is not None:
+            result['timeline'] = self._created_timeline(Sample.objects, range_param)
+
+        return result
+
+    interval_timedelta = {
+        'day': dt.timedelta(days=1),
+        'week': dt.timedelta(weeks=1),
+        'month': dt.timedelta(weeks=4),
+        'year': dt.timedelta(weeks=52)
+    }
+
+    interval_timestep = {
+        'day': dt.timedelta(hours=1),
+        'week': dt.timedelta(days=1),
+        'month': dt.timedelta(days=2),
+        'year': dt.timedelta(weeks=4)
+    }
+
+    def _get_time_intervals(self, range_param):
+        current_date = dt.datetime.now(tz=timezone.utc)
+        time_step = self.interval_timestep.get(range_param)
```

Comment:
This is a bit of weird pattern, but it's okay, just move the `interval_*` objects before the function definitions - however, since they're only used be a single function, you can just declare them at the start of the function